packages/app-desktop/gui/Sidebar/listItemComponents/FolderItem.tsx

```diff
@@ -0,0 +1,100 @@
+import { connect } from 'react-redux';
+
+const renderFolderIcon = (folderIcon: FolderIcon) => {
+	if (!folderIcon) {
+		const defaultFolderIcon: FolderIcon = {
+			dataUrl: '',
+			emoji: '',
+			name: 'far fa-folder',
+			type: FolderIconType.FontAwesome,
+		};
+		return <div style={{ marginRight: 7, display: 'flex' }}><FolderIconBox opacity={0.7} folderIcon={defaultFolderIcon}/></div>;
+	}
+
+	return <div style={{ marginRight: 7, display: 'flex' }}><FolderIconBox folderIcon={folderIcon}/></div>;
+};
+
+interface FolderItemProps {
+	themeId: number;
+	hasChildren: boolean;
+	showFolderIcon: boolean;
+	isExpanded: boolean;
+	parentId: string;
+	depth: number;
+	folderId: string;
+	folderTitle: string;
+	folderIcon: FolderIcon;
+	selectedFolderId: string;
+	notesParentType: string;
+	noteCount: number;
+	onFolderDragStart_: ItemDragListener;
+	onFolderDragOver_: ItemDragListener;
+	onFolderDrop_: ItemDragListener;
+	itemContextMenu: ItemContextMenuListener;
+	folderItem_click: (folderId: string)=> void;
+	onFolderToggleClick_: ItemClickListener;
+	shareId: string;
+	selected: boolean;
+	anchorRef: React.Ref<HTMLElement>;
+}
+
+function FolderItem(props: FolderItemProps) {
+	const { hasChildren, showFolderIcon, isExpanded, parentId, depth, selected, folderId, folderTitle, folderIcon, noteCount, onFolderDragStart_, onFolderDragOver_, onFolderDrop_, itemContextMenu, folderItem_click, onFolderToggleClick_, shareId } = props;
+
+	const noteCountComp = noteCount ? <StyledNoteCount className="note-count-label">{noteCount}</StyledNoteCount> : null;
+	const shareIcon = shareId && !parentId ? <StyledShareIcon className="fas fa-share-alt"></StyledShareIcon> : null;
+	const draggable = ![getTrashFolderId(), Folder.conflictFolderId()].includes(folderId);
+
+	const doRenderFolderIcon = () => {
+		if (folderId === getTrashFolderId()) {
+			return renderFolderIcon(getTrashFolderIcon(FolderIconType.FontAwesome));
+		}
+
+		if (!showFolderIcon) return null;
+		return renderFolderIcon(folderIcon);
+	};
+
+	return (
+		<StyledListItem depth={depth} selected={selected} className={`list-item-container list-item-depth-${depth} ${selected ? 'selected' : ''}`} onDragStart={onFolderDragStart_} onDragOver={onFolderDragOver_} onDrop={onFolderDrop_} draggable={draggable} data-folder-id={folderId}>
+			<ExpandLink hasChildren={hasChildren} folderId={folderId} onClick={onFolderToggleClick_} isExpanded={isExpanded}/>
+			<StyledListItemAnchor
+				ref={props.anchorRef}
+				className="list-item"
+				isConflictFolder={folderId === Folder.conflictFolderId()}
+				href="#"
+				selected={selected}
+				shareId={shareId}
+				data-id={folderId}
+				data-type={ModelType.Folder}
+				onContextMenu={itemContextMenu}
+				data-folder-id={folderId}
+				onClick={() => {
+					folderItem_click(folderId);
+				}}
+				onDoubleClick={onFolderToggleClick_}
+			>
+				{doRenderFolderIcon()}<StyledSpanFix className="title" style={{ lineHeight: 0 }}>{folderTitle}</StyledSpanFix>
```

Comment:
Setting `lineHeight` to zero is making writing Playwright tests for this difficult (Playwright considers text with lineHeight 0 to be invisible). It was originally added in [this commit](https://github.com/laurent22/joplin/commit/e97bb78ce442b335084de9001df63fd29e1ed079#diff-e512c98e4f56ee020ee8f2a8d5abb0b60023aa9944c74c1a89dcdfefbfdd2d27R108). Removing it fixes `expect(...).toBeVisible()` and seems to have no visible effect.